src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/FileBasedProgramsProjectSystem.cs

```diff
@@ -30,10 +31,12 @@ internal sealed class FileBasedProgramsProjectSystem : LanguageServerProjectLoad
     private readonly ILspServices _lspServices;
     private readonly ILogger<FileBasedProgramsProjectSystem> _logger;
     private readonly IMetadataAsSourceFileService _metadataAsSourceFileService;
+    private readonly VirtualProjectXmlProvider _projectXmlProvider;
```

Comment:
At this point I am thinking just placing this stuff in a `FileBasedPrograms` namespace and thinking of some distinctive names will be better. Naming this `VirtualFileBasedProgramCSharpProjectProvider` or some such.. It is so much to scan past, and it makes keeping track of the code elements harder, in my opinion. It feels important to be able to distinguish this easily from a "project system" and "project factory" and so on. There are many code elements in these layers which are all called "project" yet which are very different things.